DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs

```diff
@@ -123,19 +138,29 @@ bool registerJourneyContainsTermsAndConditions
             DelegateRegistrationModel delegateRegistrationModel,
             int userId,
             int delegateId,
-            DateTime currentTime
+            DateTime currentTime,
+            PossibleEmailUpdate emailUpdate
         )
         {
             connection.EnsureOpen();
             var transaction = connection.BeginTransaction();
 
-            userDataService.SetCentreEmail(
-                userId,
-                delegateRegistrationModel.Centre,
-                delegateRegistrationModel.CentreSpecificEmail,
-                null,
-                transaction
-            );
+            if (emailUpdate.IsEmailUpdating)
+            {
+                var emailVerified =
+                    string.IsNullOrWhiteSpace(delegateRegistrationModel.CentreSpecificEmail) ||

```

Comment:
I'm a bit concerned that this is the only place we care about whether the email is null or whitespace. Shouldn't we do this always/never? (Never = `NewEmailIsVerified` should already consider whether the email is null or whitespace)